rust/publish-zenodo/src/lib.rs

```diff
@@ -0,0 +1,575 @@
+    /// tokens" section. Give the token a name and enable the "deposit:actions" the scope.
+    /// 
+    /// Enable the "deposit:write" scope to enable the `--force` flag.
+    #[arg(long, env = "STENCILA_ZENODO_TOKEN")]
+    #[arg(help_heading("Zenodo Settings"))]
+    #[arg(display_order(1))]
+    token: Option<String>,
+
+    // Server selection options
+
+    /// Publish to Zenodo's testing server
+    #[arg(group = "zenodo_server")]
+    #[arg(long, default_value_t = true)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    sandbox: bool,
+
+    /// Publish to Zenodo's public-facing server
+    /// 
+    /// [default]
+    #[arg(group = "zenodo_server")]
+    #[arg(long)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    zenodo: bool,
+
+    /// Publish to a specific Zenodo server
+    #[arg(group = "zenodo_server")]
+    #[arg(long, value_parser = parse_host)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    server: Option<url::Host>,
+
+    // Resource type options
+
+    /// Upload document as a "lesson"
+    #[arg(group = "resource_type")]
+    #[arg(long)]
+    #[arg(conflicts_with_all = ["publication", "publication_type"])]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    lesson: bool,
+
+    /// Upload document as a "publication"
+    /// 
+    /// [default]
+    #[arg(group = "resource_type")]
+    #[arg(long, default_value_t = true)]
+    #[arg(default_value_if("lesson", common::clap::builder::ArgPredicate::IsPresent, "false"))]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    publication: bool,
+
+    /// Reserve a DOI for the deposition
+    #[arg(long)]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    reserve_doi: bool,
+
+    /// Publication date
+    /// 
+    /// When omitted, Zenodo will use today's date.
+    #[arg(long)]
+    #[arg(help_heading("Deposition Metadata"), display_order(3))]
+    #[arg(value_parser = parse_date)]
+    publication_date: Option<schema::Date>,
+
+    /// Title to use for the deposit
+    /// 
+    /// Required when the information is not available within the document.
+    #[arg(long)]
+    #[arg(help_heading("Deposition Metadata"), display_order(3))]
+    title: Option<String>,
+
+    /// Description to use within the deposition
+    /// 
+    /// Required when the information is not available within the document.
+    #[arg(long)]
+    #[arg(help_heading("Deposition Metadata"), display_order(3))]
+    description: Option<String>,
+
+    ///  One of the publication types from Zenodo's controlled vocabulary.
+    /// 
+    /// [default: preprint]
+    #[arg(long)]
+    #[arg(required_if_eq("publication", "true"))]
+    #[arg(default_value_if("publication", "true", "preprint"))]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    publication_type: Option<PublicationType>,
+
+    /// Force publish the deposition immediately
```

Comment:
add:

(use with care)

Reply:
I've updated the help text.

Short form:

    --force                          Publish the deposition immediately (use with care)

Long form:

```plain
--force
    Publish the deposition immediately
        
    Requires that access token provided by the `--token` option has the "deposit:write" scope.
        
    WARNING: This is permanent. It will be impossible to review the deposition or make changes
    to it before it is publicly viewable.
```